src/NHibernate.Test/NHSpecificTest/GH0831/FixtureByCode.cs

```diff
@@ -0,0 +1,237 @@
+				CanSelect(e => decimal.Divide(1.25m, e.EntityValue));
+			});
+		}
+
+		[Test]
+		public void CanHandleEquals()
+		{
+			Assert.Multiple(() =>
+			{
+				CanFilter(e => decimal.Equals(e.EntityValue, 1.0m));
+				CanFilter(e => decimal.Equals(1.0m, e.EntityValue));
+			});
+		}
+
+		[Test]
+		public void CanHandleFloor()
+		{
+			AssumeFunctionSupported("floor");
+
+			Assert.Multiple(() =>
+			{
+				CanFilter(e => decimal.Floor(e.EntityValue) > 1.0m);
+				CanSelect(e => decimal.Floor(e.EntityValue));
+			});
+		}
+
+		[Test]
+		public void CanHandleMultiply()
+		{
+			Assert.Multiple(() =>
+			{
+				CanFilter(e => decimal.Multiply(e.EntityValue, 10m) > 10m);
+				CanFilter(e => decimal.Multiply(10m, e.EntityValue) > 10m);
+
+				CanSelect(e => decimal.Multiply(e.EntityValue, 10m));
+				CanSelect(e => decimal.Multiply(10m, e.EntityValue));
+			});
+		}
+
+		[Test]
+		public void CanHandleNegate()
+		{
+			Assert.Multiple(() =>
+			{
+				CanFilter(e => decimal.Negate(e.EntityValue) > -1.0m);
+				CanSelect(e => decimal.Negate(e.EntityValue));
+			});
+		}
+
+		[Test]
+		public void CanHandleRemainder()
+		{
+			Assume.That(TestDialect.SupportsModuloOnDecimal, Is.True);
+
+			Assert.Multiple(() =>
+			{
+				CanFilter(e => decimal.Remainder(e.EntityValue, 2m) == 0);
+				CanFilter(e => decimal.Remainder(2m, e.EntityValue) < 1);
+
+				CanSelect(e => decimal.Remainder(e.EntityValue, 2m));
+				CanSelect(e => decimal.Remainder(2m, e.EntityValue));
+			});
+		}
+
+		[Test]
+		public void CanHandleRound()
+		{
+			AssumeFunctionSupported("round");
+
+			Assert.Multiple(() =>
+			{
+				CanFilter(e => decimal.Round(e.EntityValue) >= 2.0m);
+				CanFilter(e => decimal.Round(e.EntityValue, 1) >= 1.5m);
+
+				// SQL round() always rounds up. 
+				CanSelect(e => decimal.Round(e.EntityValue), entities.Select(e => decimal.Round(e.EntityValue, MidpointRounding.AwayFromZero)));
+				CanSelect(e => decimal.Round(e.EntityValue, 1), entities.Select(e => decimal.Round(e.EntityValue, 1, MidpointRounding.AwayFromZero)));
+			});
+		}
+
+		[Test]
+		public void CanHandleSubtract()
+		{
+			Assert.Multiple(() =>
+			{
+				CanFilter(e => decimal.Subtract(e.EntityValue, 1m) > 1m);
+				CanFilter(e => decimal.Subtract(2m, e.EntityValue) > 1m);
+
+				CanSelect(e => decimal.Subtract(e.EntityValue, 1m));
+				CanSelect(e => decimal.Subtract(2m, e.EntityValue));
+			});
+		}
+
+		private void CanFilter(Expression<Func<Entity, bool>> predicate)
+		{
+			using (ISession session = OpenSession())
+			using (session.BeginTransaction())
+			{
+				IEnumerable<Entity> inMemory = entities.Where(predicate.Compile()).ToList();
+				IEnumerable<Entity> inSession = session.Query<Entity>().Where(predicate).ToList();
+
+				CollectionAssert.AreEquivalent(inMemory, inSession);
+			}
+		}
+
+		private void CanSelect(Expression<Func<Entity, decimal>> predicate)
+		{
+			IEnumerable<decimal> inMemory = entities.Select(predicate.Compile()).ToList();
```

Comment:
If you wish to keep on explicitly ordering the in-memory case, add the ordering here:
`entities.OrderBy(e => e.EntityValue).Select(predicate.Compile()).ToList()`.

Reply:
No. I would like to get rid of it entirely. It is noise. It should be replaced with an assertion that uses a tolerance when comparing values and where the order does not matter. As far as I can find, NUnit doesn't have this. It can either assert with a tolerance, or where the order doesn't matter, not both.

I noticed that `NHibernate.Test` already has a `ObjectAssertion`, `NHAssert` and `SubclassAssert`. I can add one for collections.